lib/name_drop/client.rb

```diff
@@ -89,10 +89,15 @@ def delete(endpoint)
     # @return [Hash]
     # @raise [NameDrop::Error] Rescuing RestClient::ExceptionWithResponse
     def request(method, endpoint, attributes = {})
-      response = RestClient::Request.execute(create_request_hash(method, endpoint, attributes))
-      JSON.parse(response) unless response.empty?
+      safe_json_parse(RestClient::Request.execute(create_request_hash(method, endpoint, attributes)))
     rescue RestClient::ExceptionWithResponse => err
-      raise NameDrop::Error.new(error_message(method), JSON.parse(err.response).with_indifferent_access)
+      raise NameDrop::Error.new(error_message(method), safe_json_parse(err.response))
```

Comment:
We would need to check the response code here for 404, as there are lots of other errors that could be returned from the request, and everywhere else, we wouldn't want to check for 404, just pass it along, the only place we want to check for 404 is in the find.